packages/react/src/lib/renderComponent.tsx

```diff
@@ -144,98 +146,86 @@ const renderComponent = <P extends {}>(config: RenderConfig<P>): React.ReactElem
-          ...callable(mergedStyles.root)(styleParam),
-          ...animationCSSProp,
-        }
-
-        const resolvedStyles: ComponentSlotStylesPrepared = Object.keys(mergedStyles).reduce(
-          (acc, next) => ({ ...acc, [next]: callable(mergedStyles[next])(styleParam) }),
-          {},
-        )
-
-        const classes: ComponentSlotClasses = getClasses(renderer, mergedStyles, styleParam)
-        classes.root = cx(className, classes.root, props.className)
-
-        const config: RenderResultConfig<P> = {
-          ElementType,
-          unhandledProps,
-          classes,
-          variables: resolvedVariables,
-          styles: resolvedStyles,
-          accessibility,
-          rtl,
-          theme,
-        }
-
-        if (accessibility.focusZone) {
-          return renderWithFocusZone(render, accessibility.focusZone, config, focusZoneRef)
-        }
-
-        return render(config)
-      }}
-    </FelaTheme>
+  if (_.isEmpty(theme)) {
+    logProviderMissingWarning()
+  }
+
+  const {
+    siteVariables = {
+      colorScheme: {},
+      colors: {},
+      contextualColors: {},
+      emphasisColors: {},
+      naturalColors: {},
+      fontSizes: {},
+    },
+    componentVariables = {},
+    componentStyles = {},
+    rtl = false,
+    renderer = felaRenderer,
+  } = theme || {}
+  const ElementType = getElementType({ defaultProps }, props) as React.ReactType<P>
+
+  const stateAndProps = { ...state, ...props }
+
+  // Resolve variables for this component, allow props.variables to override
+  const resolvedVariables: ComponentVariablesObject = mergeComponentVariables(
+    componentVariables[displayName],
+    props.variables,
+  )(siteVariables)
+
+  const animationCSSProp = props.animation ? createAnimationStyles(props.animation, theme) : {}
+
+  // Resolve styles using resolved variables, merge results, allow props.styles to override
+  const mergedStyles: ComponentSlotStylesPrepared = mergeComponentStyles(
+    componentStyles[displayName],
+    {
+      root: props.styles,
+    },
+  )
+
+  const accessibility: AccessibilityBehavior = getAccessibility(stateAndProps, actionHandlers, rtl)
+
+  const unhandledProps = getUnhandledProps({ handledProps }, props)
+
+  const colors = generateColorScheme(stateAndProps.color, resolvedVariables.colorScheme)
+
+  const styleParam: ComponentStyleFunctionParam = {
+    props: stateAndProps,
+    variables: resolvedVariables,
+    theme,
+    colors,
+  }
+
+  mergedStyles.root = {
+    ...callable(mergedStyles.root)(styleParam),
+    ...animationCSSProp,
+  }
+
+  const resolvedStyles: ComponentSlotStylesPrepared = Object.keys(mergedStyles).reduce(
+    (acc, next) => ({ ...acc, [next]: callable(mergedStyles[next])(styleParam) }),
+    {},
   )
+
+  const classes: ComponentSlotClasses = getClasses(renderer, mergedStyles, styleParam)
+  classes.root = cx(className, classes.root, props.className)
+
+  const resultConfig: RenderResultConfig<P> = {
```

Comment:
There is no real changes is this code expect rename `config` to `resultConfig` as we already have `config` in this function.

Reply:
nit: should we name it as `resolvedConfig` or something similar, as, otherwise, it is a bit hard to reason what `result` stands for. However, just a suggestion, feel free to decide :)